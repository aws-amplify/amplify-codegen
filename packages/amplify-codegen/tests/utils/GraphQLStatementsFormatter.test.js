const { GraphQLStatementsFormatter } = require('../../src/utils'); 

describe('GraphQL statements Formatter', () => {
  const statements = new Map();
  statements.set('getTodo', `
    query GetProject($id: ID!) {
      getProject(id: $id) {
        id
        name
        createdAt
        updatedAt
      }
    }
  `);

  it('Generates formatted output for JS frontend', () => {
    const formattedOutput = (new GraphQLStatementsFormatter('javascript')).format(statements);
    expect(formattedOutput).toMatchSnapshot();
  });

  it('Generates formatted output for TS frontend', () => {
    const formattedOutput = (new GraphQLStatementsFormatter('typescript')).format(statements);
    expect(formattedOutput).toMatchSnapshot();
  });

  it('Generates formatted output for Flow frontend', () => {
    const formattedOutput = (new GraphQLStatementsFormatter('flow')).format(statements);
    expect(formattedOutput).toMatchSnapshot();
  });

  it('Generates formatted output for Angular frontend', () => {
    const formattedOutput = (new GraphQLStatementsFormatter('angular')).format(statements);
    // Note that for Angular, we generate in GraphQL language itself.
    expect(formattedOutput).toMatchSnapshot();
  });

  it('Generates formatted output for GraphQL language', () => {
    const formattedOutput = (new GraphQLStatementsFormatter('graphql')).format(statements);
    expect(formattedOutput).toMatchSnapshot();
  });
});
