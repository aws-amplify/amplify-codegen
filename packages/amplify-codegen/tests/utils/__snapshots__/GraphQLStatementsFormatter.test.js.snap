// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL statements Formatter Generates formatted output for Angular frontend 1`] = `
"# this is an auto generated file. This will be overwritten

query GetProject($id: ID!) {
  getProject(id: $id) {
    id
    name
    createdAt
    updatedAt
  }
}
"
`;

exports[`GraphQL statements Formatter Generates formatted output for Flow frontend 1`] = `
"// @flow
// this is an auto generated file. This will be overwritten

export const getProject = /* GraphQL */ \`
  query GetProject($id: ID!) {
    getProject(id: $id) {
      id
      name
      createdAt
      updatedAt
    }
  }
\`;
"
`;

exports[`GraphQL statements Formatter Generates formatted output for GraphQL language 1`] = `
"# this is an auto generated file. This will be overwritten

query GetProject($id: ID!) {
  getProject(id: $id) {
    id
    name
    createdAt
    updatedAt
  }
}
"
`;

exports[`GraphQL statements Formatter Generates formatted output for JS frontend 1`] = `
"/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const getProject = /* GraphQL */ \`
  query GetProject($id: ID!) {
    getProject(id: $id) {
      id
      name
      createdAt
      updatedAt
    }
  }
\`;
"
`;

exports[`GraphQL statements Formatter Generates formatted output for TS frontend 1`] = `
"/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

import * as APITypes from \\"../API\\";
type GeneratedQuery<InputType, OutputType> = string & {
  __generatedQueryInput: InputType;
  __generatedQueryOutput: OutputType;
};

export const getProject = /* GraphQL */ \`query GetProject($id: ID!) {
  getProject(id: $id) {
    id
    name
    createdAt
    updatedAt
  }
}
\` as GeneratedQuery<
  APITypes.GetProjectQueryVariables,
  APITypes.GetProjectQuery
>;
"
`;
